### PR TITLE
Refactor launch task

### PR DESCRIPTION
**Pull request by scrosby**
## Changes proposed in this PR

- Refactor launch task so it occurs behind the ComputeCluster API.
- 
-

### CODE REVIEW

File: scheduler/src/cook/mesos/task.clj [hunk 1]
```diff
@@ -63,7 +63,7 @@
            (cook-executor-candidate? job-ent))))
 
 ; TODO: This is mesos specific.
-(defn build-executor-environment
```
Comment (pschorf) [inline scheduler/src/cook/mesos/task.clj]: This is probably not actually mesos specific, the kubernetes progress reporter will probably use the same environment variables for backwards compatibility.

File: scheduler/src/cook/mesos/task.clj [hunk 2]
```diff
@@ -78,16 +78,42 @@
...
+  "Extract the executor key value from the job"
+  [db job-ent]
+  (let [container (util/job-ent->container db job-ent)
         ;; If the custom-executor attr isn't set, we default to using a custom
         ;; executor in order to support jobs submitted before we added this field
         custom-executor? (use-custom-executor? job-ent)
+        cook-executor? (use-cook-executor? job-ent)]
+        ;; executor-key configure whether this is a command or custom executor
+        (cond
+                       (and container cook-executor?) :container-cook-executor
```
Comment (pschorf) [inline scheduler/src/cook/mesos/task.clj]: nitpick: indentation looks weird here.

File: scheduler/src/cook/mesos/task.clj (hunk 1 above)
Comment (scrosby) [reply scheduler/src/cook/mesos/task.clj]: Ok. Name reverted.

File: scheduler/src/cook/mesos/task.clj (hunk 2 above)
Comment (scrosby) [reply scheduler/src/cook/mesos/task.clj]: Good catch.

**Comment by pschorf** [comment]
It's still marked wip: can you clear that and confirm that the internal build is green?

**Pull request merged**